compiler-plugin-test/src/test/kotlin/it/unibo/collektive/test/util/CompileUtils.kt

```diff
@@ -0,0 +1,93 @@
+package it.unibo.collektive.test.util
+
+import com.tschuchort.compiletesting.JvmCompilationResult
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import io.kotest.matchers.string.shouldNotContain
+import it.unibo.collektive.AlignmentComponentRegistrar
+import org.apache.commons.text.StringSubstitutor
+import org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi
+import java.util.HashMap
+
+@OptIn(ExperimentalCompilerApi::class)
+object CompileUtils {
+    fun compile(fileName: String, program: String): JvmCompilationResult {
+        val sourceFile = SourceFile.kotlin(fileName, program)
+
+        return KotlinCompilation().apply {
+            sources = listOf(sourceFile)
+            compilerPluginRegistrars = listOf(AlignmentComponentRegistrar())
+            inheritClassPath = true
+        }.compile()
+    }
+
+    data class KotlinTestingProgram(
+        val fileName: String,
+        private val template: String,
+        val program: String,
+        private val properties: Map<String, String>,
+    ) {
+
+        fun formatCode(vararg args: Any?): KotlinTestingProgram =
+            KotlinTestingProgram(fileName, template, program.format(*args), properties)
+
+        fun put(key: String, value: String): KotlinTestingProgram {
+            val updateProperties = properties + (key to value)
+            return KotlinTestingProgram(
+                fileName,
+                template,
+                StringSubstitutor.replace(template, updateProperties, "%(", ")"),
+                updateProperties,
+            )
+        }
+
+        fun import(javaClass: Class<*>): KotlinTestingProgram =
+            put(
+                "imports",
+                properties["imports"] + "\nimport " + javaClass.name,
+            )
+
+        infix fun shouldCompileWith(compilationCheck: (JvmCompilationResult) -> Unit) {
+            println(program)
+            println(properties)
```

Comment:
Are all these `println` needed or are they something from your experiments?

Reply:
I'm removing them right now, sorry. They are not needed.